crates/ruff_linter/src/importer/mod.rs

```diff
@@ -135,7 +135,20 @@ impl<'a> Importer<'a> {
             self.stylist,
         )?;
 
+        // Add the import to a `TYPE_CHECKING` block.
+        if let Some(block) = self.preceding_type_checking_block(at) {
+            // Add the import to the existing `TYPE_CHECKING` block.
+            return Ok(TypingImportEdit {
+                type_checking_edit: None,
+                add_import_edit: self.add_to_type_checking_block(&content, block.start()),
+            });
+        }
+
         // Import the `TYPE_CHECKING` symbol from the typing module.
+        // TODO: Should we provide an option to avoid this import?
+        //       E.g. either through an explicit setting, or implicitly
+        //       when `typing` isn't part of the exempt modules and there
+        //       are no other existing runtime imports of `typing`.
```

Comment:
Hmm, yeah this is interesting. Not having an option seems fine to me for now, we can introduce one later if it is being asked for.